### PR TITLE
[batch] Populate billing project users by date deduped table

### PR DESCRIPTION
**Pull request by jigold**
Stacked on #12813

### CODE REVIEW

**Review by daniel-goldstein — APPROVED** [review]
This looks great. Read through the audit twice and didn't notice anything

**Comment by jigold** [comment]
@daniel-goldstein Can you take a look at the last commit?

File: batch/sql/dedup_billing_project_users_by_date.py [hunk 1]
```diff
@@ -158,12 +158,13 @@ async def audit_changes(db):
   LOCK IN SHARE MODE
 ) AS old
 LEFT JOIN (
-  SELECT billing_date, billing_project, `user`, resource_id, CAST(COALESCE(SUM(`usage`), 0) AS SIGNED) AS `usage`
+  SELECT billing_date, billing_project, `user`, deduped_resource_id, CAST(COALESCE(SUM(`usage`), 0) AS SIGNED) AS `usage`
   FROM aggregated_billing_project_user_resources_by_date_v3
+  LEFT JOIN resources ON resources.resource_id = aggregated_billing_project_user_resources_by_date_v3.resource_id
```
Comment (daniel-goldstein) [inline batch/sql/dedup_billing_project_users_by_date.py:163]: Just double-checking, this LEFT JOIN is intentionally resource_id not deduped?

**Review by daniel-goldstein — CHANGES_REQUESTED** [review]

File: batch/sql/dedup_billing_project_users_by_date.py (hunk 1 above)
Comment (jigold) [reply batch/sql/dedup_billing_project_users_by_date.py:163]: I'm pretty sure this is correct. The `v3` table doesn't have a `deduped_resource_id` to even join on.

**Review by daniel-goldstein — APPROVED** [review]

**Pull request merged**